models/event_argument_extraction/KnowledgeExtractor.py

```diff
@@ -0,0 +1,79 @@
+from transformers import AutoModelForSeq2SeqLM, AutoTokenizer
+
+class KnowledgeExtractor(object):
+    def __init__(self):
+        self.path_to_ckpt = ''
+        self.tokenizer = None
+        self.model = None
+        self.model_para = dict()
+
+    def instantiate(self):
+        pass
+         
+    def forward(self):
+        pass
+ 
+class RebelKnowledgeExtractor(KnowledgeExtractor):
+
+    def instantiate(self, model="Babelscape/rebel-large", max_length = 256, length_penalty = 0, num_beams = 3, num_return_sequences = 3):
+        self.tokenizer = AutoTokenizer.from_pretrained(model)
+        self.model = AutoModelForSeq2SeqLM.from_pretrained(model)
+        self.model_para = {"max_length": max_length,
+                                        "length_penalty": length_penalty,
+                                        "num_beams": num_beams,
+                                        "num_return_sequences": num_return_sequences,
```

Comment:
```suggestion
    def instantiate(self, path_to_ckpt="Babelscape/rebel-large"):
        tokenizer = AutoTokenizer.from_pretrained(model)
        model = AutoModelForSeq2SeqLM.from_pretrained(model)
        return tokenizer, model
```

Reply:
it is generally better to explicitly output instances(tokenizer and model), instead of implicitly assigning the instances to variables defined in the constructor.

Reply:
done